chia/introducer/introducer.py

```diff
@@ -49,28 +50,33 @@ async def _vetting_loop(self):
                 if len(raw_peers) == 0:
                     continue
 
+                peer: VettedPeer
                 for peer in raw_peers:
                     if self._shut_down:
                         return
-                    if peer.get_hash() in self.vetted_timestamps:
-                        if time.time() > self.vetted_timestamps[peer.get_hash()] + 3600:
-                            if peer.get_hash() in self.vetted:
-                                self.vetted[peer.get_hash()] = False
-                    if peer.get_hash() not in self.vetted or not self.vetted[peer.get_hash()]:
-                        try:
-                            self.log.info(f"Vetting peer {peer.host} {peer.port}")
-                            r, w = await asyncio.wait_for(
-                                asyncio.open_connection(peer.host, int(peer.port)),
-                                timeout=3,
-                            )
-                            w.close()
-                        except Exception as e:
-                            self.log.warning(f"Could not vet {peer}. {type(e)}{str(e)}")
-                            self.vetted[peer.get_hash()] = False
-                            continue
 
-                        self.log.info(f"Have vetted {peer} successfully!")
-                        self.vetted[peer.get_hash()] = True
-                        self.vetted_timestamps[peer.get_hash()] = int(time.time())
+                    # if it was too long ago we checked this peer, check it
+                    # again
+                    if peer.vetted and time.time() > peer.vetted_timestamp + 3600:
+                        peer.vetted = False
+
+                    if peer.vetted:
+                        continue
+
+                    try:
+                        self.log.info(f"Vetting peer {peer.host} {peer.port}")
+                        r, w = await asyncio.wait_for(
+                            asyncio.open_connection(peer.host, int(peer.port)),
+                            timeout=3,
+                        )
+                        w.close()
+                    except Exception as e:
+                        self.log.warning(f"Could not vet {peer}, removing. {type(e)}{str(e)}")
+                        self.server.introducer_peers.remove(peer)
```

Comment:
as far as I could tell, we would *never* remove peers previously. I made this change that if we fail to connect to a peer, we remove it.

Reply:
the peer might be temporary offline (and then we loose it forever when it reconnects). perhaps remove if time.time() > peer.vetted_timestamp + 24 * 3600?

Reply:
Yeah, that's a good point

Reply:
if vetting has *never* passed for a peer, `vetted_timestamp` would be zero, causing the peer to be booted immediately. Instead of comparing the timestamp, I count the number of consecutive failures and remove the peer once there are too many. Also, to avoid vetting a peer too frequently, I added another timestamp member to skip vetting if we did it recently

Reply:
My understanding is that we'll drop all of this once we have a working DNS introducer, right?
I think there are lots of improvements to make here, but I don't think it makes sense if we're dropping it soon,

Reply:
yes, this will be replaced soon with better peer selection logic included.